components/board/pi/impl/servo.go

```diff
@@ -109,7 +109,7 @@ func (s *piPigpioServo) Move(ctx context.Context, angle uint32, extra map[string
 		angle = s.max
```

Comment:
I'm sorry, I just noticed a design flaw in this approach.

Right now, `servo.max` and `servo.min` is used as a software limit to a default angle range. 
Concretely, our range is 0-180 and if `min` is set to 10 and `max` is set to `150` we're only allowing a user to move between those ranges.

If we set our max as our range limit, then we can never set a maximum angle within our range through software.. This changes the user-facing behavior of that config parameter. 

We may need to add a config parameter something like `range`, or change the min-max configs to only be used in the pulse calculation, and not in limiting the min and max behaviour.

Reply:
Oh so min and max are meant to be angle sizes not parameters of the servo and the code is assuming all servos move between 0 and 180 degrees?

Reply:
Yes, so we need a third parameter and a new item on that servo struct `range` for example (you can pick a better name), default that to a 180 angle, and then if a user sets a `range` in the config we use that instead of our default 180 angle.

The way to do that would be to add (in `components/board/pi/common/common.go`):
```
	Range       int      `json:"range,omitempty"` 
```
The omit empty means it is option, and defaults to a `0` when our attributes convert it. We check that  that in the constructor and if it is non-zero, we set the servo range to that number, as you were doing with `max`.

We'd also have to add a check where a user adds a range that is smaller than max. That can be done in `Vaidate`